### DISCUSSION TITLE
HelmRelease not ready and HelmChart downloads

### DISCUSSION THREAD

**Discussion by avthart** [fluxcd/flux2#739] [Q&A] [answered]
We have deployed e.g. cert-manager (v1.1.0) using a HelmRelease. Deployed and running successfully. 

After a while, the HelmChart could not be downloaded because of an (temporary) HTTP 503 error. This currently leads to a not ready status for the HelmRelease, even when there 's no change (not an updated/newer version) for that HelmRelease. 

```
NAME                                                           CHART           VERSION          SOURCE KIND      SOURCE NAME     READY   STATUS                                                                                                AGE
helmchart.source.toolkit.fluxcd.io/flux-system-cert-manager    cert-manager    v1.1.x           HelmRepository   jetstack        False   failed to fetch https://charts.jetstack.io/charts/cert-manager-v1.1.0.tgz : 503 Service Unavailable   23h

NAME                                               READY   STATUS                                                          AGE
helmrelease.helm.toolkit.fluxcd.io/cert-manager    False   HelmChart 'flux-system/flux-system-cert-manager' is not ready   23h
```

Is this what we want and expected behaviour? It looks like HelmCharts will be downloaded even when that version is already downloaded before.

**Comment by stefanprodan** [comment, marked as answer]
Yes, since chart versions are not guaranteed to be immutable, we download them every time. 

PS. You can control the interval at which source-controller tries to download the chart with `HelmRelease.spec.chart.spec.interval`, more details here https://toolkit.fluxcd.io/components/helm/helmreleases/
